### ISSUE TITLE
getAttributeRawValue() - when call with array of attrs unexpectable behavior

### ISSUE THREAD

**Issue by Sekiphp** [OpenMage/magento-lts#658]
**Class:** Mage_Catalog_Model_Resource_Abstract

**Example call:**
```
$productData = Mage::getResourceModel('catalog/product')->getAttributeRawValue(
                    (int) $suborder->getProductId(),
                    ['package_height', 'package_width', 'package_length', 'weight'],
                    Mage_Core_Model_App::ADMIN_STORE_ID
                );
```

**Expected result:** indexed array
```
Array
(
    [weight] => 53.5000
    [package_height] => 0
    [package_width] => 0
    [package_length] => 0
)
```

**Given result:** `53.5000` (Because only one of given attributes exists for this product in admin store.)

**Comment by colinmollenhour** [comment]
I thought something similar came up before? It seems the correct approach is to look at the original number of arguments to determine whether or not to return a scalar or an array, not the result for reasons such as this.

**Comment by midlan** [comment]
@colinmollenhour I've created fix (pull request #664), where I look if the list of attributes is array. If yes, I disabled conversion to scalar.

**Comment by Sekiphp** [comment]
Solved in pull request #664
